launcher/container_runner_test.go

```diff
@@ -408,54 +354,73 @@ func testRetryPolicyWithNTries(t *testing.T, numTries int, expectRefresh bool) {
 		}
 		return nil, errors.New("attest unsuccessful")
 	}
+
+	tokenWriter := models.NewFakeTokenWriter()
+	fakeTimer := models.NewFakeTimer()
 	runner := ContainerRunner{
 		attestAgent: &fakeAttestationAgent{attestFunc: attestFunc},
 		logger:      logging.SimpleLogger(),
+		tokenWriter: tokenWriter,
```

Comment:
I think the implementation could just be the function signature
```
fetchAndWriteTokenWithRetry(ctx context.Context,
	retry func() *backoff.ExponentialBackOff, refresh func(ctx context.Context) (time.Duration, error)) error
```

That way you can just pass in a `FakeRefresher` with a `Refresh` and a `GetNextToken` method.